util/include/Metrics.hpp

```diff
@@ -29,6 +30,14 @@ class Gauge;
 class Status;
 class Counter;
 
+// A generic struct that may represent a counter or a gauge
+// the motivation is to eliminate that need to know the exact
+// metric name before getting it from the aggregator
+typedef struct {
+  std::string component;
+  std::string name;
+  uint64_t value;
```

Comment:
I talked to @yontyon offline and he's going to change the type of 'value' to return `std::variant<Status, Counter, Gauge, Histogram>` so we don't lose context and can return non-numeric types.

Reply:
done